Fixes #2919

jtgrassie opened the pull request:
Add checks for termcap if needed by readline

danrmiller commented:
Thanks. Might be a little frustrating!

https://build.getmonero.org/builders/monero-static-ubuntu-amd64/builds/3079/steps/compile/logs/stdio
```
-- Found Readline: /usr/include  
-- Looking for rl_copy_text
-- Looking for rl_copy_text - not found
-- Looking for rl_filename_completion_function
-- Looking for rl_filename_completion_function - not found
-- Could not find GNU readline library so building without readline support
```
https://build.getmonero.org/builders/monero-static-ubuntu-amd64/builds/3079/steps/compile/logs/CMakeError
```
Determining if the function rl_copy_text exists failed with the following output:
Change Dir: /home/vagrant/slave/monero-static-ubuntu-amd64/build/build/release/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTC_f5bc9/fast"
make[1]: Entering directory '/home/vagrant/slave/monero-static-ubuntu-amd64/build/build/release/CMakeFiles/CMakeTmp'
/usr/bin/make -f CMakeFiles/cmTC_f5bc9.dir/build.make CMakeFiles/cmTC_f5bc9.dir/build
make[2]: Entering directory '/home/vagrant/slave/monero-static-ubuntu-amd64/build/build/release/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_f5bc9.dir/CheckFunctionExists.c.o
/usr/bin/cc    -fno-strict-aliasing -maes -std=c11 -D_GNU_SOURCE   -Wall -Wextra -Wpointer-arith -Wundef -Wvla -Wwrite-strings -Wno-error=extra -Wno-error=deprecated-declarations -Wno-unused-parameter -Wno-unused-variable -Wno-error=unused-variable -Wno-error=undef -Wno-error=uninitialized -Wlogical-op -Wno-error=maybe-uninitialized -Wno-error=cpp -Waggregate-return -Wnested-externs -Wold-style-definition -Wstrict-prototypes -march=x86-64   -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=rl_copy_text   -o CMakeFiles/cmTC_f5bc9.dir/CheckFunctionExists.c.o   -c /usr/share/cmake-3.5/Modules/CheckFunctionExists.c
/usr/share/cmake-3.5/Modules/CheckFunctionExists.c:6:1: warning: function declaration isn't a prototype [-Wstrict-prototypes]
 char CHECK_FUNCTION_EXISTS();
 ^
Linking C executable cmTC_f5bc9
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_f5bc9.dir/link.txt --verbose=1
/usr/bin/cc   -fno-strict-aliasing -maes -std=c11 -D_GNU_SOURCE   -Wall -Wextra -Wpointer-arith -Wundef -Wvla -Wwrite-strings -Wno-error=extra -Wno-error=deprecated-declarations -Wno-unused-parameter -Wno-unused-variable -Wno-error=unused-variable -Wno-error=undef -Wno-error=uninitialized -Wlogical-op -Wno-error=maybe-uninitialized -Wno-error=cpp -Waggregate-return -Wnested-externs -Wold-style-definition -Wstrict-prototypes -march=x86-64   -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=rl_copy_text    CMakeFiles/cmTC_f5bc9.dir/CheckFunctionExists.c.o  -o cmTC_f5bc9 -rdynamic /usr/lib/x86_64-linux-gnu/libreadline.a 
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_get_screen_size':
(.text+0x201): undefined reference to `tgetnum'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_get_screen_size':
(.text+0x2cb): undefined reference to `tgetnum'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x647): undefined reference to `PC'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x65a): undefined reference to `BC'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x665): undefined reference to `UP'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x6f3): undefined reference to `tgetent'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x72f): undefined reference to `tgetstr'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x75b): undefined reference to `PC'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x771): undefined reference to `BC'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x77f): undefined reference to `UP'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x78f): undefined reference to `tgetflag'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x7d0): undefined reference to `tgetflag'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x832): undefined reference to `tgetflag'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x8fb): undefined reference to `tgetent'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_backspace':
(.text+0xad5): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `rl_ding':
(.text+0xb8f): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_enable_meta_key':
(.text+0xbf5): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_disable_meta_key':
(.text+0xc47): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_control_keypad':
(.text+0xc7b): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o):(.text+0xcbe): more undefined references to `tputs' follow
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2418): undefined reference to `tgoto'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x242a): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2c26): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2e23): undefined reference to `tgoto'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2e33): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2ea1): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2ebd): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2eec): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `rl_redisplay':
(.text+0x4509): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o):(.text+0x481c): more undefined references to `tputs' follow
collect2: error: ld returned 1 exit status
```
I'm not sure what info is needed:
```
$ ls -la /usr/lib/x86_64-linux-gnu/libncurses.a /usr/lib/x86_64-linux-gnu/libreadline.a /usr/lib/x86_64-linux-gnu/libtermcap.a
-rw-r--r-- 1 root root 298812 Feb 19  2016 /usr/lib/x86_64-linux-gnu/libncurses.a
-rw-r--r-- 1 root root 545496 Feb  4  2016 /usr/lib/x86_64-linux-gnu/libreadline.a
lrwxrwxrwx 1 root root     10 Feb 19  2016 /usr/lib/x86_64-linux-gnu/libtermcap.a -> libtinfo.a
```
```
$ nm -A /usr/lib/x86_64-linux-gnu/*.a | egrep 'tgetnum' 
/usr/lib/x86_64-linux-gnu/libreadline.a:terminal.o:                 U tgetnum
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_termcap.o:0000000000000630 T tgetnum
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_termcap.o:0000000000000630 T tgetnum
```
```
$ nm -A /usr/lib/x86_64-linux-gnu/*.a | egrep 'tputs'  
/usr/lib/x86_64-linux-gnu/libcurses.a:lib_color.o:                 U tputs
/usr/lib/x86_64-linux-gnu/libcurses.a:lib_mvcur.o:                 U tputs
/usr/lib/x86_64-linux-gnu/libcurses.a:lib_vidattr.o:                 U tputs
/usr/lib/x86_64-linux-gnu/libcurses.a:tty_update.o:                 U tputs
/usr/lib/x86_64-linux-gnu/libncurses.a:lib_color.o:                 U tputs
/usr/lib/x86_64-linux-gnu/libncurses.a:lib_mvcur.o:                 U tputs
/usr/lib/x86_64-linux-gnu/libncurses.a:lib_vidattr.o:                 U tputs
/usr/lib/x86_64-linux-gnu/libncurses.a:tty_update.o:                 U tputs
/usr/lib/x86_64-linux-gnu/libreadline.a:display.o:                 U tputs
/usr/lib/x86_64-linux-gnu/libreadline.a:terminal.o:                 U tputs
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:                 U __ctype_b_loc
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:                 U cur_term
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:00000000000000b0 T delay_output
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:                 U __errno_location
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:                 U fileno
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:                 U getenv
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:                 U _IO_putc
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:                 U napms
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:                 U _nc_baudrate
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:0000000000000040 T _nc_flush
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:0000000000000000 B _nc_nulls_sent
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:0000000000000160 T _nc_outch
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:                 U _nc_prescreen
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:0000000000000000 T _nc_putchar
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:0000000000000540 T _nc_putp
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:0000000000000010 T _nc_set_no_padding
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:0000000000000004 B ospeed
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:0000000000000006 B PC
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:0000000000000530 T putp
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:                 U SP
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:                 U __stack_chk_fail
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:0000000000000000 d static_outch
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:                 U stdout
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:                 U strchr
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:0000000000000230 T tputs
/usr/lib/x86_64-linux-gnu/libtermcap.a:lib_tputs.o:                 U write
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:                 U __ctype_b_loc
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:                 U cur_term
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:00000000000000b0 T delay_output
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:                 U __errno_location
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:                 U fileno
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:                 U getenv
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:                 U _IO_putc
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:                 U napms
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:                 U _nc_baudrate
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:0000000000000040 T _nc_flush
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:0000000000000000 B _nc_nulls_sent
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:0000000000000160 T _nc_outch
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:                 U _nc_prescreen
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:0000000000000000 T _nc_putchar
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:0000000000000540 T _nc_putp
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:0000000000000010 T _nc_set_no_padding
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:0000000000000004 B ospeed
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:0000000000000006 B PC
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:0000000000000530 T putp
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:                 U SP
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:                 U __stack_chk_fail
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:0000000000000000 d static_outch
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:                 U stdout
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:                 U strchr
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:0000000000000230 T tputs
/usr/lib/x86_64-linux-gnu/libtinfo.a:lib_tputs.o:                 U write
```

moneromooo-monero commented on cmake/FindReadline.cmake:
```diff
@@ -19,56 +19,57 @@
+  NAMES readline/readline.h
+  PATHS ${Readline_ROOT_DIR}/include
+  NO_DEFAULT_PATH
 )
```
All these seem to be unchanged apart from whitespace. Please refrain.

jtgrassie replied:
@danrmiller the build bot doesn't have libreadline. So that's expected output.

@moneromooo-monero yes I know. Apologies. I find editing cmake files a PITA and re-indenting helped me. The file is so small I hoped this wouldn't hide the actual changes in review.

jtgrassie replied:
@danrmiller I stand corrected! It looks like something else. Let me try adding tinfo.

danrmiller commented:
https://build.getmonero.org/builders/monero-static-ubuntu-amd64/builds/3084

https://build.getmonero.org/builders/monero-static-ubuntu-amd64/builds/3084/steps/compile/logs/CMakeError

```
Determining if the function rl_copy_text exists failed with the following output:
Change Dir: /home/vagrant/slave/monero-static-ubuntu-amd64/build/build/release/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTC_7645a/fast"
make[1]: Entering directory '/home/vagrant/slave/monero-static-ubuntu-amd64/build/build/release/CMakeFiles/CMakeTmp'
/usr/bin/make -f CMakeFiles/cmTC_7645a.dir/build.make CMakeFiles/cmTC_7645a.dir/build
make[2]: Entering directory '/home/vagrant/slave/monero-static-ubuntu-amd64/build/build/release/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_7645a.dir/CheckFunctionExists.c.o
/usr/bin/cc    -fno-strict-aliasing -maes -std=c11 -D_GNU_SOURCE   -Wall -Wextra -Wpointer-arith -Wundef -Wvla -Wwrite-strings -Wno-error=extra -Wno-error=deprecated-declarations -Wno-unused-parameter -Wno-unused-variable -Wno-error=unused-variable -Wno-error=undef -Wno-error=uninitialized -Wlogical-op -Wno-error=maybe-uninitialized -Wno-error=cpp -Waggregate-return -Wnested-externs -Wold-style-definition -Wstrict-prototypes -march=x86-64   -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=rl_copy_text   -o CMakeFiles/cmTC_7645a.dir/CheckFunctionExists.c.o   -c /usr/share/cmake-3.5/Modules/CheckFunctionExists.c
/usr/share/cmake-3.5/Modules/CheckFunctionExists.c:6:1: warning: function declaration isn't a prototype [-Wstrict-prototypes]
 char CHECK_FUNCTION_EXISTS();
 ^
Linking C executable cmTC_7645a
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_7645a.dir/link.txt --verbose=1
/usr/bin/cc   -fno-strict-aliasing -maes -std=c11 -D_GNU_SOURCE   -Wall -Wextra -Wpointer-arith -Wundef -Wvla -Wwrite-strings -Wno-error=extra -Wno-error=deprecated-declarations -Wno-unused-parameter -Wno-unused-variable -Wno-error=unused-variable -Wno-error=undef -Wno-error=uninitialized -Wlogical-op -Wno-error=maybe-uninitialized -Wno-error=cpp -Waggregate-return -Wnested-externs -Wold-style-definition -Wstrict-prototypes -march=x86-64   -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=rl_copy_text    CMakeFiles/cmTC_7645a.dir/CheckFunctionExists.c.o  -o cmTC_7645a -rdynamic /usr/lib/x86_64-linux-gnu/libreadline.a 
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_get_screen_size':
(.text+0x201): undefined reference to `tgetnum'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_get_screen_size':
(.text+0x2cb): undefined reference to `tgetnum'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x647): undefined reference to `PC'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x65a): undefined reference to `BC'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x665): undefined reference to `UP'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x6f3): undefined reference to `tgetent'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x72f): undefined reference to `tgetstr'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x75b): undefined reference to `PC'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x771): undefined reference to `BC'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x77f): undefined reference to `UP'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x78f): undefined reference to `tgetflag'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x7d0): undefined reference to `tgetflag'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x832): undefined reference to `tgetflag'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x8fb): undefined reference to `tgetent'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_backspace':
(.text+0xad5): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `rl_ding':
(.text+0xb8f): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_enable_meta_key':
(.text+0xbf5): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_disable_meta_key':
(.text+0xc47): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_control_keypad':
(.text+0xc7b): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o):(.text+0xcbe): more undefined references to `tputs' follow
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2418): undefined reference to `tgoto'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x242a): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2c26): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2e23): undefined reference to `tgoto'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2e33): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2ea1): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2ebd): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2eec): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `rl_redisplay':
(.text+0x4509): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o):(.text+0x481c): more undefined references to `tputs' follow
collect2: error: ld returned 1 exit status
```

jtgrassie commented:
@danrmiller 
https://bugzilla.redhat.com/show_bug.cgi?id=499837

The problem is knowing which termcap library any specific version of readline is built against it seems.

jtgrassie commented:
I believe this is build bot host related rather than anything else. Without have access to the actual host it's tough to see what's installed regarding readline and its dependencies.
It's not new or related specifically to this PR (which provides a fairly good improvement to cmake readline and termcap searching & linking).

jtgrassie commented:
Oh cmake. Right, I tested on ubuntu 16 and found that the function checks were not re-running with same variable names used on the second run. Fixed and squashed.

moneromooo-monero commented:
If this is ready to review now, please fix all the arbitrary whitespace and case changes.

jtgrassie commented:
@moneromooo-monero done and ready

danrmiller commented:
Yes this also works on osx boxes with the OS libedit and brew's libreadline installed.

ston1th commented:
Just tested this on OpenBSD. Works like a charm :+1:

moneromooo-monero commented:
iDunk reports "I always forget to comment on #2978 that it needs include(CheckFunctionExists) in cmake/FindReadline.cmake."

jtgrassie commented:
Something else is already including it which is why I omitted. 
I can duplicate here if you prefer though.

jtgrassie commented:
@moneromooo-monero
> iDunk reports "I always forget to comment on #2978 that it needs include(CheckFunctionExists) in cmake/FindReadline.cmake."

OK I have added the include. It does make the module more portable.

mmortal03 commented:
I don't know if it matters, but was this also fixed when building for Windows, with make debug-static-win64 in MSYS2? I have the following packages installed, to be sure:
libreadline, mingw-w64-x86_64-readline, and mingw-w64-x86_64-libreadline-devel 

I'm seeing the following, up through at least 3fb4223. I don't know why it doesn't find them, then searches again and finds them, but ultimately thinks it could not find them:

-- Could NOT find Readline (missing: Readline_INCLUDE_DIR)
-- Looking for rl_copy_text
-- Looking for rl_copy_text - not found
-- Looking for rl_filename_completion_function
-- Looking for rl_filename_completion_function - not found
-- Looking for rl_copy_text
-- Looking for rl_copy_text - found
-- Looking for rl_filename_completion_function
-- Looking for rl_filename_completion_function - found
-- Could not find GNU readline library so building without readline support
